engine/functions_tests/nested2.yaml

```diff
@@ -22,6 +22,9 @@ conditions:
         - value: '{{.step.this.state}}'
           operator: EQ
           expected: 'NESTED2_STATE1'
+        - value: '{{ .function_args.name1 }}'
+          operator: EQ
+          expected: foobar
```

Comment:
I added this condition in the test : function `nested2` has correctly been called with `foobar`, but it's failing in the test.
This comes from the fact that the pull request is only using the function args of the "main" step (the one declared in the template): if we start to have nested functions, templating start to fails, because only the `function_args` on the "main" step is populated in the values.

I also found why this test was failing if we used `values.Clone`, follow the second comment